lib/spack/spack/build_environment.py

```diff
@@ -404,12 +404,15 @@ def set_build_environment_variables(pkg, env, dirty):
 
 def _set_variables_for_single_module(pkg, module):
     """Helper function to set module variables for single module."""
-    # number of jobs spack will build with.
-    jobs = spack.config.get('config:build_jobs') or multiprocessing.cpu_count()
+
+    # The package required a serial build by setting parallel=False
     if not pkg.parallel:
-        jobs = 1
-    elif pkg.make_jobs:
-        jobs = pkg.make_jobs
+        spack.config.set('config:build_jobs', 1, scope='_builtin')
+
+    # Number of jobs Spack will build with
+    jobs = spack.config.get(
+        'config:build_jobs', default=multiprocessing.cpu_count()
+    )
```

Comment:
I'm not a fan of changing a global variable, just to read it two lines later.  It brings up the question of... what what is the scope of the `_builtin` config scope, and where else in Spack will this cause an effect?  Could this logic be rewritten as follows?  I think it's easier to understand.  (And if the rewrite is wrong; that tells me the existing logic is hard to understand).
```
jobs = spack.config.get(
    'config:build_jobs', default=multiprocessing.cpu_count()) \
    if pkg.parallel else 1
```